ruddock/modules/perm_mgr/helpers.py

```diff
@@ -0,0 +1,187 @@
+import flask
+import sqlalchemy
+from ruddock.resources import Permissions
+
+def fetch_office_permissions():
```

Comment:
I mean I suppose lol, does it matter?

Reply:
ahh, i tried it out with more thorough test data, and now i see it. you wanna show every office that's got permissions, even if they're not active, in case you need to remove them. likewise, you want to show all active offices, even if they have no permissions, in case you want to manage them.

same with users.

if that's the case, lemme suggest the following (you gotta have the extra any_permissions column because there's a bug where you can't sort on expressions involving aliases: https://bugs.mysql.com/bug.php?id=80802)

`SELECT GROUP_CONCAT(permission_id) AS permissions,
`  permission_id IS NOT NULL AS any_permissions,
`  office_name, office_id, office_order, is_active
`FROM offices
`  NATURAL LEFT JOIN office_permissions
`GROUP BY office_id
`HAVING is_active OR any_permissions
`ORDER BY
`  any_permissions DESC, -- puts nulls after non-nulls
`  permissions,
`  office_order

Reply:
Hmm, it worked fine on my machine (using MariaDB, maybe they fixed it?). I checked and the Ruddock website is using MariaDB also so this might not be needed?

Reply:
I'm also using mariadb; maybe I'm just on an older version. In any case, I think that approach makes it much clearer what you're intending to fetch there.